From 3028208762a975f90dd9f08b6e62fe224290269b Mon Sep 17 00:00:00 2001
From: Emil Renner Berthing <systemd@esmil.dk>
Date: Thu, 18 Sep 2014 15:24:51 +0200
Subject: [PATCH 07/17] shared/utmp-wtmp: don't fail if libc doesn't support
 utmpx/wtmpx

I'd like to make utmp/wtmp handling a configure option, but for
now this is the minimal change needed to make it compile with musl.
The musl utmp/wtmp functions doesn't do anything anyway.

Upstream-Status: Inappropriate [musl specific]

Signed-off-by: Andre McCurdy <armccurdy@gmail.com>
---
 src/shared/utmp-wtmp.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/shared/utmp-wtmp.c b/src/shared/utmp-wtmp.c
index 9750dcd..a9018e0 100644
--- a/src/shared/utmp-wtmp.c
+++ b/src/shared/utmp-wtmp.c
@@ -72,8 +72,10 @@ int utmp_get_runlevel(int *runlevel, int *previous) {
                 return 0;
         }
 
+#ifdef _PATH_UTMPX
         if (utmpxname(_PATH_UTMPX) < 0)
                 return -errno;
+#endif
 
         setutxent();
 
@@ -131,8 +133,10 @@ static int write_entry_utmp(const struct utmpx *store) {
          * each entry type resp. user; i.e. basically a key/value
          * table. */
 
+#ifdef _PATH_UTMPX
         if (utmpxname(_PATH_UTMPX) < 0)
                 return -errno;
+#endif
 
         setutxent();
 
@@ -153,7 +157,9 @@ static int write_entry_wtmp(const struct utmpx *store) {
         simply appended to the end; i.e. basically a log. */
 
         errno = 0;
+#ifdef _PATH_WTMPX
         updwtmpx(_PATH_WTMPX, store);
+#endif
         return -errno;
 }
 
-- 
1.9.1

